src/pages/PullRequestPage/subroute/ComponentsSelector/ComponentsSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import isUndefined from 'lodash/isUndefined'
+import { useState } from 'react'
+
+import { useLocationParams } from 'services/navigation'
+import { usePullComponents } from 'services/pull'
+import { useFlags } from 'shared/featureFlags'
+import Icon from 'ui/Icon'
+import MultiSelect from 'ui/MultiSelect'
+
+const defaultQueryParams = {
+  search: '',
+  components: [],
+}
+
+function ComponentsSelector() {
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
+  const [componentSearch, setComponentSearch] = useState('')
+  const [selectedComponents, setSelectedComponents] = useState(
+    // @ts-expect-errors, useLocation params needs to be updated to have full types
+    params?.components
+  )
+  const { componentsSelect: componentsSelectFlag } = useFlags({
+    componentsSelect: false,
+  })
+  const { data, isLoading } = usePullComponents({
+    filters: {
+      components: componentSearch ? [componentSearch] : undefined,
+    },
+    options: {
+      suspense: false,
+    },
+  })
+
+  const components =
+    data?.pull?.compareWithBase.__typename === 'Comparison'
```

Comment:
Curious what this means. Do components only exist in this scenario?

Reply:
Yes! We only have results if compareWithBase is of comparison type, we have other union types surfaced as errors like `MissingBaseCommit` and `MissingHeadCommit` but these are already handled in the hook level: https://github.com/codecov/gazebo/blob/main/src/services/pull/usePullComponents.tsx

More on union types [here](https://graphql.org/learn/schema/#union-types)